tests/discrete_space/test_grid_sampling.py

```diff
@@ -0,0 +1,84 @@
+"""Unit tests for Grid.select_random_empty_cell heuristic behavior."""
```

Comment:
please include these in the overall test file rather than adding new test files.